src/transports/sentry_transport_winhttp.c

```diff
@@ -110,6 +111,21 @@ static int
 sentry__winhttp_transport_shutdown(uint64_t timeout, void *transport_state)
 {
     sentry_bgworker_t *bgworker = (sentry_bgworker_t *)transport_state;
+    winhttp_bgworker_state_t *state = sentry__bgworker_get_state(bgworker);
+
+    if (state->connect) {
+        WinHttpCloseHandle(state->connect);
```

Comment:
These are also being closed in `state_free`.

Reply:
Yes if everything goes well. In case some WinHttp functions are taking longer than the timeout the background thread will still be alive. Closing the handle will cancel the  WinhttpReceiveResponse call and unblocks the background thred.